JavaScript/JavaScriptSDK.Tests/CheckinTests/SendBuffer.tests.ts

```diff
@@ -0,0 +1,420 @@
+
+        this.testCase({
+            name: "ArraySendBuffer: initialize",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // verify
+                Assert.equal(0, buffer.count(), "new buffer should be empty");
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: can enqueue and clear the buffer",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // act
+                buffer.enqueue("");
+
+                // verify
+                Assert.equal(1, buffer.count(), "one item expected");
+
+                // act
+                buffer.enqueue("");
+
+                // verify
+                Assert.equal(2, buffer.count(), "two items expected");
+
+                //act
+                buffer.clear();
+
+                // verify
+                Assert.equal(0, buffer.count(), "buffer should be empty");
+
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: can clear empty buffer",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // act
+                buffer.clear();
+
+                // verify
+                Assert.equal(0, buffer.count(), "buffer should be empty");
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: call batchPayloads when a buffer is empty",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // act
+                var batch = buffer.batchPayloads();
+
+                // verify
+                Assert.equal(null, batch, "expecting null");
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: call batchPayloads when a buffer has one element",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // act
+                var payload = "{ test: test }";
+
+                buffer.enqueue(payload);
+                var batch = buffer.batchPayloads();
+
+                // verify
+                Assert.equal("[" + payload + "]", batch, "invalid batch");
+            }
+        });
+
+        this.testCase({
+            name: "ArraySendBuffer: call batchPayloads when a buffer has two elements",
+            test: () => {
+                // setup
+                var buffer = this.getArraySendBuffer();
+
+                // act
+                var payload1 = "{ test: test }";
+                var payload2 = "{ }";
+
+                buffer.enqueue(payload1);
+                buffer.enqueue(payload2);
+                var batch = buffer.batchPayloads();
+
+                // verify
+                Assert.equal("[" + payload1 + "," + payload2 + "]", batch, "invalid batch");
```

Comment:
What are the benefits of using this trick? String concatenation is expensive and table join will work similar to `StringBuilder `from C#? 

TS compiler doesn't optimize this code. Both versions, before and after the compilation look exactly the same.